File: external/naver-client/src/main/java/com/news/feign/NaverClientConfiguration.java
package com.news.feign;

import com.fasterxml.jackson.databind.ObjectMapper;
import feign.RequestInterceptor;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;

public class NaverClientConfiguration {
    @Bean
    public RequestInterceptor requestInterceptor(
            @Value("${external.naver.headers.client-id}") String naverClientId,
            @Value("${external.naver.headers.client-secret}") String naverClientSecret
    ){
        return requestTemplate -> requestTemplate.header("X-Naver-Client-Id", naverClientId)
                .header("X-Naver-Client-Secret", naverClientSecret);
    }

    @Bean
    public NaverErrorDecoder naverErrorDecoder(ObjectMapper objectMapper){
        return new NaverErrorDecoder(objectMapper);
    }
}

File: external/naver-client/src/main/kotlin/com/search/feign/NaverClientConfiguration.kt
package com.search.feign

import com.fasterxml.jackson.databind.ObjectMapper
import feign.RequestInterceptor
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

class NaverClientConfiguration {

    @Bean
    fun requestInterceptor(
            @Value("\${external.naver.headers.client-id}") naverClientId: String,
            @Value("\${external.naver.headers.client-secret}") naverClientSecret: String
    ) : RequestInterceptor{
        return RequestInterceptor { requestTemplate ->
             requestTemplate.header("X-Naver-Client-Id", naverClientId)
             requestTemplate.header("X-Naver-Client-Secret", naverClientSecret)
        }
    }

    @Bean
    fun naverErrorDecoder(objectMapper: ObjectMapper) : NaverErrorDecoder{
        return NaverErrorDecoder(objectMapper)
    }
}